### DISCUSSION TITLE
Nuxt is very vague about the configuration method of postcss. How to configure the postcss plug-in and declare the configuration content?

### DISCUSSION THREAD

**Discussion by xhc-code** [nuxt/nuxt#20725] [Questions] [answered]
Nuxt is very vague about the configuration method of postcss. How to configure the postcss plug-in and declare the configuration content?

From the api documentation, it is impossible to see where the postcss is configured and where the corresponding configuration options are passed.
![image](https://user-images.githubusercontent.com/28866025/236820827-f54487b0-915c-4dc0-8c48-1cd1827a5e27.png)

My understanding is:
```
postcss: {
    plugins: {
      plugin_name: plugin_options object
   }
}
```

https://nuxt.com/docs/api/configuration/nuxt-config#postcss

**Comment by TheAlexLichter** [comment, marked as answer]
Hey 👋 

From the Nuxt types:

```ts
postcss: {
  /**
   * Options for configuring PostCSS plugins.
   * 
   * https://postcss.org/
   * 
  */
  plugins: Record<string, any>,
 },
```

So, yes, something like:

```ts
postcss: {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
    pluginName: { /*plugin options*/ }
  }
```

Feel free to improve the [TS docs](https://github.com/nuxt/nuxt/blob/a793e7ad48c56a034fd547081ed32b47a810c23e/packages/schema/src/config/postcss.ts#L7C7-L11) to make it more clear (part of the docs is generated through it)😋 
PR welcome!

**Reply by xhc-code** [reply]
![image](https://user-images.githubusercontent.com/28866025/236975181-24d850b7-1b3f-4d81-81c6-0b0df76402d0.png)
Official website document example plug-in here, you see that these two categories are inconsistent, what is the usage expressed?

**Reply by TheAlexLichter** [reply]
This is resolved from https://github.com/nuxt/nuxt/blob/a793e7ad48c56a034fd547081ed32b47a810c23e/packages/schema/src/config/postcss.ts#L13-L49.  It is actually incomplete. Will check 👍

**Reply by xhc-code** [reply]
ok
